src/api/trace/span_context.rs

```diff
@@ -111,6 +114,110 @@ impl SpanId {
     }
 }
 
+/// TraceState carries system-specific configuration data, represented as a list
+/// of key-value pairs. TraceState allows multiple tracing systems to
+/// participate in the same trace.
+///
+/// Please review the [W3C specification] for details on this field.
+///
+/// [W3C specification]: https://www.w3.org/TR/trace-context/#tracestate-header
+#[cfg_attr(feature = "serialize", derive(Deserialize, Serialize))]
+#[derive(Clone, Debug, Default, Eq, PartialEq)]
+pub struct TraceState {
+    data: HashMap<String, String>,
```

Comment:
If this is an ordered map, could it simply be a `BTreeMap`?

Reply:
The problem is that the sorting is by insertion order, not key. So we could just keep everything in a `Vec` but then you have to iterate through everything on `get`. Rust doesn't have something like [Boost multi index](https://www.boost.org/doc/libs/1_62_0/libs/multi_index/doc/index.html), which would be ideal. There's an upper bound on how long trace state can be in the spec, so maybe it's not a big deal to just make this a wrapper around a `VecDeque` and `get` can just be `O(n)` where `n` isn't that big.